### PR TITLE
[HttpKernel] Fix wrapping the end of handleException() on PHP 7

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

I discovered this while working on https://github.com/symfony/symfony/pull/33065.

https://github.com/symfony/symfony/pull/458 is not working with PHP 7 if the listener throws an `\Error`. That means we should catch `\Throwable` here.

### CODE REVIEW

**Review by yceruto — DISMISSED** [review]

**Comment by nicolas-grekas** [comment]
This code is here since a long time, for good reasons to me: `Error` *should not* take the same path as `Exception`. They were created as a separate type hierarchy *because* the deserve a dedicated separate code path. Here the dev made a mistake and must fix it asap.

**Comment by nicolas-grekas** [comment]
TL;DR: never swallow an `Error`.

**Comment by fancyweb** [comment]
@nicolas-grekas explanation makes sense. Let's close it.
